docs/source/architecture/sub_stakes.rst

```diff
@@ -0,0 +1,270 @@
+Sub-stakes
+==========
+
+A staker may extend the unlock period for any number of portions of their total stake. This divides the stake into smaller parts, each with a unique unlock date in the future. Stakers may also acquire and lock new tokens. The total stake is represented as the sum of all the different sub-stakes active in a given cycle (new cycle every 24h), which includes locked sub-stakes, and any sub-stakes that have passed their unlock date, and can be freely withdrawn. Each sub-stake has a beginning and duration (lock time). When a staker confirms activity each day and winding down is enabled, the remaining lock time for relevant sub-stakes is reduced.
+
+Sub-stake remains active until become unlocked and staker get reward for last period of sub-stake by calling ``mint()`` or ``confirmActivity()`` after the last period is surpassed. Each staker can have no more than 30 active sub-stakes. All sub-stakes changes use slots of inactive sub-stakes in the first place and only after that - empty slots in the array. Therefore, obtaining data on past staker locks is not guaranteed.
```

Comment:
Few moments:
- Collecting reward is not necessary to make sub-stake inactive. We can "remove" element from sub-stakes array when we no need information about locking. After last period of sub stake become history we still need to calculate reward for this last period. It will happen inside mint() method wich work only with surpassed period.
- Not sure about "the inactive stake slots", for me sounds like slot is inactive (maybe I'm wrong). I'll try to rephrase idea: we have array with sub-stakes with fixed size (30). Some of this sub-stakes are not usefull for any calculation in StakingEscrow, they are inactive and we can reuse them. When we add new one - we just looking from left to right trying to find place where to save new element, inactive sub-stake or empty slots if there are any. Not sure that is more clear than before :)
- We allways can get information about locking for any staker for the current moment and for future, also we could try to retreive this information for some period in the past, and if were no any rewrites of inactive slots then this information still will be valid

Reply:
Thanks for the explanation - that makes things clearer. I guess the terminology `staker locks` is just a little weird to me.

```suggestion
A sub-stake remains active until it becomes unlocked, and a staker collects the reward for the last period by calling ``mint()`` or ``confirmActivity()`` once the last period is surpassed. Each staker can have no more than 30 active sub-stakes which are stored in an array. All sub-stake changes initially reuse slots of inactive sub-stakes for storage in the array, and if there are none, will instead use empty slots. Therefore, attempting to retrieve data about previous inactive sub-stakes is not guaranteed to be successful since the data could have been overwritten.
```

Reply:
Great! Just changed `collects` -> `gets`, because we have CLI command that collects reward and this command has other meaning